src/components/ViewDeck/ViewDeck.js

```diff
@@ -1,94 +1,77 @@
+  const [state, setState] = React.useState({
+    //STILL NEED STATE
+  });
+
+  const [hoverCard, setHoverCard] = useState(
+    "https://i.stack.imgur.com/Vkq2a.png"
+  );
 
-  componentDidMount() {
-    this.props.dispatch({
+  useEffect(() => {
+    props.dispatch({
       type: "GET_LIST",
-      payload: this.props.reduxStore.selectedDeck.id,
+      payload: props.reduxStore.selectedDeck.id,
     });
 
-    this.props.dispatch({
+    props.dispatch({
       type: "GET_SELECTED_DECK",
-      payload: this.props.history.location.pathname.split("/")[2],
+      payload: props.history.location.pathname.split("/")[2],
     });
-  }
+  }, [state]);
 
-  editDeck = (deck) => {
-    console.log("in edit deck with:", deck);
-    const id = this.props.history.location.pathname.split("/")[2];
-    this.props.history.push(`/editdeck/${id}`);
+  const editDeck = (deck) => {
+    const id = props.history.location.pathname.split("/")[2];
+    props.history.push(`/editdeck/${id}`);
   };
 
-  cardDisplay = (card) => {
-    //console.log('hovering on', card.name, card, card.api_data);
+  const cardDisplay = (card) => {
     var parsedData = JSON.parse(card.api_data);
-    //    console.log(parsedData.image_uris.normal);
-
     if (parsedData.image_uris === undefined) {
-      this.setState({
-        hoverCard: `https://s3.thingpic.com/images/Kz/uF3amfCnYFLBggjUNr1sPRKi.jpeg`,
-      });
+      setHoverCard(
+        `https://s3.thingpic.com/images/Kz/uF3amfCnYFLBggjUNr1sPRKi.jpeg`
+      );
     } else {
-      this.setState({
-        hoverCard: parsedData.image_uris.normal,
-      });
+      setHoverCard(parsedData.image_uris.normal);
     }
   };
-
-  removeDisplay = (card) => {
+  const removeDisplay = (card) => {
     // console.log('left:', card.name, card, card.api_data);
-    this.setState({
-      hoverCard: "https://i.stack.imgur.com/Vkq2a.png",
-    });
+    setHoverCard("https://i.stack.imgur.com/Vkq2a.png");
   };
 
-  countQty = (cards) => {
+  const countQty = (cards) => {
     if (cards) {
       var qty = 0;
       cards.forEach((card) => {
-        //for each instance of a card, qtyToAdd = card.quantity
         var qtyToAdd = card.quantity;
-        //console.log('qtyToAdd', qtyToAdd);
         let totalqty = qty + qtyToAdd;
         qty = totalqty;
-        //var totalCards = qty
-        //console.log('totalCards', totalCards);
       });
-
-      console.log("new total:", qty);
-
-      if (this.state.deckQty !== qty) {
-        this.setState({
+      if (state.deckQty !== qty) {
+        setState({
           deckQty: qty,
         });
       }
     }
   };
 
-  parseCards = (cards) => {
-    var allParsedCards = [];
-    cards.forEach((card) => {
-      var parsedCard = JSON.parse(card.api_data);
-      allParsedCards.push(parsedCard);
+  const parseCards = (cards) => {
+    return cards.map((card) => {
+      const jsonData = JSON.parse(card.api_data);
+      return {
+        ...card,
+        jsonData,
+      };
```

Comment:
The more I think about it the more I believe this should not live in the component. This should happen in a selector, or in `mapStateToProps`. Even better would be transforming the data on fetch and before putting it into the store.